### PR TITLE
Remove stof/doctrine-extensions-bundle dependency

### PR DESCRIPTION
**Pull request by franmomu**
## Subject

Ref: https://github.com/sonata-project/SonataTranslationBundle/pull/453#issuecomment-785344620

Instead of relying on the bundle, a `sonata_translation.gedmo.translatable_listener_service` option has been added to the configuration so the user must specify the translatable listener service name when using gedmo (allowing to use another bundle or only the package). 

I am targeting this branch, because these changes are BC.

## Changelog

```markdown
### Added
- Added `sonata_translation.gedmo.translatable_listener_service` option to specify the translatable listener service name.
### Deprecated
- Deprecated not setting `sonata_translation.gedmo.translatable_listener_service` option when using gedmo.
```

### CODE REVIEW

**Comment by jordisala1991** [comment]
Maybe we can generate the listener ourselves and register it if there is no value provided?

**Comment by franmomu** [comment]
I started doing that, but looking at this service definition:

https://github.com/stof/StofDoctrineExtensionsBundle/blob/f3c948b390f82a357d814331e6ea47bcda813a38/Resources/config/translatable.xml#L12

looks like it's configurable and maybe the user wants to register its own. Anyway if no service name is provided we could register a simple one, I'll change it.

File: src/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -57,6 +57,23 @@ public function getConfigTreeBuilder()
                             ->requiresAtLeastOneElement()
                             ->prototype('scalar')->end()
                         ->end()
+                        ->scalarNode('translatable_listener_service')
+                            ->isRequired()
+                            ->cannotBeEmpty()
+                            ->info('The translatable listener service that has to be registered when using gedmo/doctrine-extensions')
+                            // NEXT_MAJOR: Remove default value.
+                            ->defaultValue(static function (): string {
```
Comment (VincentLanglet) [inline src/DependencyInjection/Configuration.php:65]: I don't understand, in the doc we're saying:
```
translatable_listener_service: Gedmo\Translatable\TranslatableListener
```

Shouldn't it be the default value ?

Comment (franmomu) [reply src/DependencyInjection/Configuration.php:63]: It has to be `stof_doctrine_extensions.listener.translatable` to be BC.

**Review by jordisala1991 — APPROVED** [review]

**Review by VincentLanglet — APPROVED** [review]

**Comment by jordisala1991** [comment]
Thank you @franmomu

**Pull request merged**